code/game/machinery/computer/singulo_monitor.dm

```diff
@@ -0,0 +1,205 @@
+/obj/machinery/computer/singulo_monitor
+	name = "singularity monitoring console"
+	desc = "Used to monitor singularities."
+	icon_keyboard = "power_key"
+	icon_screen = "singumon_0"
+	circuit = /obj/item/circuitboard/singulo_monitor
+	light_color = LIGHT_COLOR_YELLOW
+	/// Cache-list of all singularities
+	var/list/singularities
+	/// Last energy level of the singularity so we know if it went up or down between cycles
+	var/last_energy
+	/// Last size of the singularity so we know if it went up or down between cycles
+	var/last_size
+	/// Reference to the active singularity
+	var/obj/singularity/active
+	/// Channel to send warning through to the engineers
+	var/warning_channel = "Engineering"
+	/// Channel to send breach containment alert
+	var/breach_channel = "Common"
+	/// Radio for sending announcements
+	var/obj/item/radio/singu_radio
+	/// List of field generators containing the singulo
+	var/list/field_gens
+
+/obj/machinery/computer/singulo_monitor/Initialize(mapload)
+	. = ..()
+	singu_radio = new(src)
+	singu_radio.listening = FALSE
+	singu_radio.follow_target = src
+	singu_radio.config(list("[warning_channel]" = 0, "[breach_channel]" = 0))
+
+
+/obj/machinery/computer/singulo_monitor/Destroy()
+	active = null
+	qdel(singu_radio)
```

Comment:
Make sure this gets emptied out as well, refs are still refs
```suggestion
	qdel(singu_radio)
	singularities.Cut()
```

Reply:
done